discovery-provider/src/queries/search_queries.py

```diff
@@ -258,69 +356,88 @@ def search(args):
-                        session, searchStr, limit, offset, True, is_auto_complete, current_user_id)
+                    submit_and_add("followed_users")
+
             if (searchKind in [SearchKind.all, SearchKind.playlists]):
-                results['playlists'] = with_users_added(playlist_search_query(
-                    session,
-                    searchStr,
-                    limit,
-                    offset,
-                    False,
-                    False,
-                    is_auto_complete,
-                    current_user_id
-                ))
+                submit_and_add("playlists")
                 if current_user_id:
-                    results['saved_playlists'] = with_users_added(playlist_search_query(
-                        session,
-                        searchStr,
-                        limit,
-                        offset,
-                        False,
-                        True,
-                        is_auto_complete,
-                        current_user_id
-                    ))
+                    submit_and_add("saved_playlists")
+
             if (searchKind in [SearchKind.all, SearchKind.albums]):
-                results['albums'] = with_users_added(playlist_search_query(
-                    session, searchStr, limit, offset, True, False, is_auto_complete, current_user_id))
+                submit_and_add("albums")
                 if current_user_id:
-                    results['saved_albums'] = with_users_added(playlist_search_query(
-                        session,
-                        searchStr,
-                        limit,
-                        offset,
-                        True,
-                        True,
-                        is_auto_complete,
-                        current_user_id
-                    ))
+                    submit_and_add("saved_albums")
 
+            for future in concurrent.futures.as_completed(futures):
+                search_result = future.result()
+                future_type = futures_map[future]
+
+                # Add to the final results
+                results[future_type] = search_result
+
+                # Add to user_ids
+                user_ids.update(get_users_ids(search_result))
+
+            with db.scoped_session() as session:
+                # Add users back
+                users = get_users_by_id(session, list(user_ids), current_user_id)
+
+                for (_, result_list) in results.items():
+                    for result in result_list:
+                        user_id = None
+                        if 'playlist_owner_id' in result:
+                            user_id = result['playlist_owner_id']
+                        elif 'owner_id' in result:
+                            user_id = result['owner_id']
+
+                        if user_id is not None:
+                            user = users[user_id]
+                            result["user"] = user
```

Comment:
I might be misunderstanding this, but does modifying this entry from `results.items` also get modified in `results` dictionary?

Reply:
It does! It's pretty cool, this method basically returns a 'view' that's kept in sync with the parent dict.

https://www.programiz.com/python-programming/methods/dictionary/items